src/Microsoft.AspNet.Mvc.Core/AntiForgery/AntiForgeryWorker.cs

```diff
@@ -204,6 +189,54 @@ public void Validate([NotNull] HttpContext httpContext, string cookieToken, stri
                 deserializedFormToken);
         }
 
+
+        /// <summary>
+        /// Generates an anti-forgery cookie token and header for this request.
+        /// </summary>
+        /// <param name="context">The HTTP context associated with the current call.</param>
+        public void SetCookieTokenAndHeader([NotNull] HttpContext httpContext)
+        {
+            CheckSSLConfig(httpContext);
+
+            var oldCookieToken = GetCookieTokenNoThrow(httpContext);
+            var newCookieToken = ValidateAndGenerateNewToken(oldCookieToken);
+            
+            SaveCookieTokenAndHeader(httpContext, newCookieToken);
+        }
+
+        // This method returns null if oldCookieToken is valid.
+        private AntiForgeryToken ValidateAndGenerateNewToken(AntiForgeryToken oldCookieToken)
+        {
+            if (!_validator.IsCookieTokenValid(oldCookieToken))
+            {
+                // Need to make sure we're always operating with a good cookie token.
+                var newCookieToken = _generator.GenerateCookieToken();
+                Debug.Assert(_validator.IsCookieTokenValid(newCookieToken));
+                return newCookieToken;
+            }
+
+            return null;
+        }
+
+        private void SaveCookieTokenAndHeader(
+            [NotNull] HttpContext httpContext,
+            [NotNull] AntiForgeryToken newCookieToken)
+        {
+            if (newCookieToken != null)
```

Comment:
isnt  this check bogus? we have NotNull already. I understand the need for it but I suggest this check should be where you are calling this method.